crates/ruff_python_formatter/resources/test/fixtures/ruff/trailing_comments.py

```diff
@@ -1,5 +1,6 @@
 # Pragma reserved width fixtures
 i = ("aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",)  # noqa: This shouldn't break
+i = ("aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",)  # NoQA: This shouldn't break
```

Comment:
I'm not sure how I feel about this... Our documentation is relatively clear that it should be written all lower case. Supporting multiple casing will slow down all our tooling, for no obvious benefit. 

It also opens the question why we only support different casing for `noqa` but not `type`, `pyright`, etc...

Reply:
The answer is that it depends on what the tools themselves support, and Flake8 decided that its pragma should be case-insensitive. The same is not true of the others, even `nosec`.

We continue to support these for compatibility and I think it's important to support _at least_ `# NOQA`:

- `# noqa`: 1.7M files ([source](https://github.com/search?type=code&q=%2F%28%3F-i%29noqa%2F+language%3APython))
- `# NOQA`: 80k files ([source](https://github.com/search?type=code&q=%2F%28%3F-i%29NOQA%2F+language%3APython))
- `# NoQA`: 1.5k files ([source](https://github.com/search?type=code&q=%2F%28%3F-i%29NoQA%2F+language%3APython))

(That being said, the linter _already_ supports these so this logic should be consistent with that decision.)

Reply:
We could also consider only supporting `noqa`, `NOQA`, and `NoQA`.